ddtrace/contrib/kombu/patch.py

```diff
@@ -0,0 +1,114 @@
+# 3p
+import kombu
+import wrapt
+
+# project
+from ddtrace import config
+
+from ...pin import Pin
+from ...utils.formats import get_env
+from .constants import DEFAULT_SERVICE
+from ...ext import kombu as kombux
+from ...ext import AppTypes
+from ...utils.wrappers import unwrap
+from ...propagation.http import HTTPPropagator
+from .utils import (
+    get_exchange_from_args,
+    get_body_length_from_args,
+    get_routing_key_from_args,
+    extract_conn_tags,
+    HEADER_POS
+)
+
+# kombu default settings
+config._add('kombu',{
+    'service_name': get_env('kombu', 'service_name', DEFAULT_SERVICE)
+})
+
+propagator = HTTPPropagator()
+
+
+def patch():
+    """Patch the instrumented methods
+
+    This duplicated doesn't look nice. The nicer alternative is to use an ObjectProxy on top
+    of Kombu. However, it means that any "import kombu.Connection" won't be instrumented.
+    """
+    if getattr(kombu, '_datadog_patch', False):
+        return
+    setattr(kombu, '_datadog_patch', True)
+
+    _w = wrapt.wrap_function_wrapper
+    # We wrap the _publish method because the publish method:
+    # *  defines defaults in its kwargs
+    # *  potentially overrides kwargs with values from self
+    # *  extracts/normalizes things like exchange
+    _w(kombux.TYPE, 'Producer._publish', traced_publish)
+    _w(kombux.TYPE, 'Consumer.receive', traced_receive)
+    Pin(
+        service=config.kombu['service_name'],
+        app="kombu",
+        app_type=AppTypes.worker,
+    ).onto(kombu.messaging.Producer)
+
+    Pin(
+        service=config.kombu['service_name'],
+        app="kombu",
+        app_type=AppTypes.worker,
+    ).onto(kombu.messaging.Consumer)
+
+
+def unpatch():
```

Comment:
I'm confused about this, because I copied what Redis Contrib does, and it does not do this. In fact celery uses a "patch_app/unpatch_app" while Redis just uses patch/unpatch. Can you help me understand the difference?

Reply:
Yeah, sorry for the confusion. Our redis integration was one of the first added to the tracer and has a few issues, we're trying to correct these with our new integrations going forward 😄. 

Celery uses a `patch_app` and `unpatch_app` _in addition_ to `patch` and `unpatch` due to celery convention (I believe). Each integration must have `patch` and `unpatch` in order to be picked up by the patching mechanism.

Let me know if this all makes sense! 😄

Reply:
Since the Kombu integration doesn't have an "app" convention, should I be doing something like that? Additionally, how do I unpin in the unpatch app? I'm not sure where to get the pin to remove it. Thanks for your help!